### DISCUSSION TITLE
Setting cookies client side

### DISCUSSION THREAD

**Discussion by manan30** [vercel/next.js#49192] [Help]
### Summary

I am using the app router. I have a client component annotated with "use client" at a leaf node that is doing some data mutation. When I hit the API endpoint to do this data mutation, the API endpoint sends a "Set-cookie" header as part of the response, but the browser never seems to set it. Is this expected behavior in Next JS? If yes how can I set cookies received in client-side data fetching/mutation calls? 

P.S: Using the route handlers inside the app directory works fine and I am able to set the cookie but for direct API it seems to be breaking

**Comment by BrajBliss** [comment]
Did you find a fix?
